### PR TITLE
fix: improve error messaging

### PR DESCRIPTION
**Pull request by jackcmay**
Give a more descriptive error when a connection operation fails.

fixes: #809

### CODE REVIEW

**Comment by jackcmay** [comment]
A baby step, but better?

Next step would be to improve the error responses sent by the server

File: src/connection.js [hunk 1]
```diff
@@ -888,7 +896,12 @@ export class Connection {
     const unsafeRes = await this._rpcRequest('getAccountInfo', args);
     const res = GetAccountInfoAndContextRpcResult(unsafeRes);
     if (res.error) {
-      throw new Error(res.error.message);
+      throw new Error(
+        'failed info about account ' +
```
Comment (CriesofCarrots) [inline src/connection.js:900]:
```suggestion
        'failed to get info about account ' +
```

File: src/connection.js [hunk 2]
```diff
@@ -921,7 +934,9 @@ export class Connection {
     return await this.getAccountInfoAndContext(publicKey, commitment)
       .then(x => x.value)
       .catch(e => {
-        throw e;
+        throw new Error(
+          'failed info about account ' + publicKey.toBase58() + ': ' + e,
```
Comment (CriesofCarrots) [inline src/connection.js:938]:
```suggestion
          'failed to get info about account ' + publicKey.toBase58() + ': ' + e,
```

File: src/connection.js [hunk 3]
```diff
@@ -938,7 +953,12 @@ export class Connection {
     const unsafeRes = await this._rpcRequest('getProgramAccounts', args);
     const res = GetProgramAccountsRpcResult(unsafeRes);
     if (res.error) {
-      throw new Error(res.error.message);
+      throw new Error(
+        'failed info about account ' +
```
Comment (CriesofCarrots) [inline src/connection.js:957]:
```suggestion
        'failed to get accounts owned by program ' +
```

**Pull request merged**